Sources/eppo/EppoClient.swift

```diff
@@ -55,40 +57,71 @@ public class EppoClient {
         subjectAttributes: SubjectAttributes = SubjectAttributes(),
         defaultValue: Bool) throws -> Bool
     {
-        return try getInternalAssignment(
-            flagKey: flagKey, 
-            subjectKey: subjectKey, 
-            subjectAttributes: subjectAttributes, 
-            useTypedVariationValue: true
-        )?.boolValue() ?? defaultValue
+        do {
+            return try getInternalAssignment(
+                flagKey: flagKey, 
+                subjectKey: subjectKey, 
+                subjectAttributes: subjectAttributes,
+                expectedVariationType: UFC_VariationType.boolean
+            )?.variation?.value.getBoolValue() ?? defaultValue
+        } catch {
+            // todo: implement graceful mode
+            return defaultValue
+        }
     }
     
-    public func getJSONStringAssignment(
+    // todo: add back when supporting JSON
```

Comment:
The UFC does support JSON, transmitted as a JSON string with the variation type JSON. What our SDKs do is attempt to parse and return the native JSON root node. (E.g. no JSON type in EppoValue anymore as its represented as a string and just-in-time parsed)

Reply:
Yep understood - mostly I wanted to get the logic in this PR up for review while I figured out what to do about json serialization in swift.